server/rooms.ts

```diff
@@ -1034,6 +1034,7 @@ export class GlobalRoomState {
 	maxUsers: number;
 	maxUsersDate: number;
 	formatList: string;
+	writingData?: boolean;
```

Comment:
```suggestion
	pendingDataWrite: NodeJS.Timer | null = null;
```

the rest is left as an exercise for the reader